cascade-merge/action.yml

```diff
@@ -67,57 +74,68 @@ runs:
             dest=${CASCADE_BRANCHES[$((${step}+1))]}
             git checkout $dest
             msg="$dest <= $src"
-            git merge -m "$msg" $src || (echo " === $msg KO ==="; exit 1)
+            git merge -m "$msg" $src || (echo " === $msg KO ==="; echo "SOME_CONFLICTS=true" >> $GITHUB_ENV)
             echo " === $msg OK ==="
         done
       shell: bash
 
-#   waiting for slack secrets to be shared as organization secrets:
-#    https://ledgerhq.atlassian.net/servicedesk/customer/portal/14/ACCESS-3920?created=true
-#    - name: Notify failing merge
-#      if: failure()
-#      uses: LedgerHQ/actions/slack-notification@main
-#      env:
-#        SLACK_WEBHOOK_URL: ${{ inputs.slack-webhook-url }}
-#        SLACK_API_TOKEN: ${{ inputs.slack-api-token }}
-#        SLACK_CUSTOM_PAYLOAD: |-
-#          {
-#            "blocks": [
-#              {
-#                "type": "header",
-#                "text": {
-#                  "type": "plain_text",
-#                  "text": ":boom: Merge conflict :boom:",
-#                  "emoji": true
-#                }
-#              },
-#              {
-#                "type": "section",
-#                "text": {
-#                  "type": "mrkdwn",
-#                  "text": "<@SLACK_MEMBER_ID> please fix it :iamwatchingyou:"
-#                }
-#              },
-#              {
-#                "type": "actions",
-#                "elements": [
-#                  {
-#                    "type": "button",
-#                    "text": {
-#                      "type": "plain_text",
-#                      "text": ":github: Github build",
-#                      "emoji": true
-#                    },
-#                    "url": "https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}"
-#                  }
-#                ]
-#              },
-#              {
-#                "type": "divider"
-#              }
-#            ]
-#          }
-#
+    - name: Check if slack creds are provided
+      if: always()
+      run: |
+        if [ "${{ env.SOME_CONFLICTS }}" == "true" ] && [ "${{ inputs.slack-webhook-url }}" != "" ]; then
+          echo "SHOULD_SEND_SLACK_NOTIF=true" >> $GITHUB_ENV
+        fi
+      shell: bash
+
+    - name: Notify failing merge
+      if: ${{ env.SHOULD_SEND_SLACK_NOTIF == 'true' }}
```

Comment:
there is no "if always()" here, will it run if one of the previous step fail?
Or it works because it is *after* a `if: always` ?
(sorry not gh action expert)

BTW, maybe we can merge the 2 steps

Reply:
actually, there is no failing happening at this step
the workaround I did to run slack notification optionally if there slack creds are provided is to do the following:
- `echo "SOME_CONFLICTS=true" >> $GITHUB_ENV` instead of `exit 1`
- and then the following step, I exit if `SOME_CONFLICTS` is true
- `if: always()` in previous step, can be removed, it is useless. because there is no failing happening.
- anyway, I tested everything in all cases, and it was working normally.